dimod/generators/mimo.py

```diff
@@ -0,0 +1,1085 @@
+
+        amps: Amplitudes as an iterable.
+
+        quadrature: Quadrature (True) or only phase-shift keying such as BPSK (False).
+
+        random_state: Seed for a random state or a random state.
+
+    Returns:
+
+        Symbols, as a column vector of length ``num_transmitters``.
+
+    """
+
+    if np.iscomplex(amps).any():
+        raise ValueError('Amplitudes cannot have complex values')
+
+    if any(np.modf(amps)[0]):
+        raise ValueError('Amplitudes must have integer values')
+
+    random_state = _make_random_state(random_state)
+
+    if quadrature == False:
+        transmitted_symbols = random_state.choice(amps, size=(num_transmitters, 1))
+    else:
+        transmitted_symbols = random_state.choice(amps, size=(num_transmitters, 1)) \
+            + 1j*random_state.choice(amps, size=(num_transmitters, 1))
+
+    return transmitted_symbols
+
+def _create_signal(F, 
+                   transmitted_symbols=None, 
+                   channel_noise=None, 
+                   SNRb=float('Inf'), 
+                   modulation='BPSK', 
+                   channel_power=None, 
+                   random_state=None):
+    """Simulate a transmission signal. 
+
+    Generates random transmitted symbols and optionally noise, math:`y = F v + n`,
+    where the channel, :math:`F`, is assumed to consist of independent and 
+    identically distributed (i.i.d) elements such that 
+    :math:`F\dagger*F = N_r I[N_t]*cp` where :math:`I` is the identity matrix 
+    and :math:`cp` the channel power; the transmitted symbols, :math:`v` or 
+    ``transmitted_symbols``, are assumed to consist of i.i.d unscaled 
+    constellations elements (integer valued in real and complex parts), and 
+    mean constellation power dictates a rescaling relative to 
+    :math:`E[v v\dagger] = I[Nt]`.
+
+    ``channel_noise`` is assumed, or created, to be suitably scaled: 
+    :math:`N_0 I[Nt] = SNRb `.
+
+    Args:
+        F: Wireless channel as an :math:`i \times j` matrix of complex values,
+            where :math:`i` rows correspond to :math:`y_i` receivers and :math:`j`
+            columns correspond to :math:`v_i` transmitted symbols.
+
+        transmitted_symbols: Transmitted symbols as a column vector.
+
+        channel_noise: Channel noise as a complex value.
+
+        SNRb: Signal-to-noise ratio.
+
+        modulation: Modulation. Supported values are 'BPSK', 'QPSK', '16QAM',
+            '64QAM', and '256QAM'.
+
+        channel_power: Channel power. By default, proportional to the number
+            of transmitters.
+
+        random_state: Seed for a random state or a random state.
+
+    Returns:
+        Four-tuple of received signals (``y``), transmitted symbols (``v``),
+        channel noise, and random_state, where ``y`` is a column vector of length
+        equal to the rows of ``F``.
+    """
+
+    num_receivers = F.shape[0]
+    num_transmitters = F.shape[1]
+
+    random_state = _make_random_state(random_state)
```

Comment:
```suggestion
    rng = np.random.default_rng(seed)
```
to be consistent with NumPy and other (modern) parts of Ocean.